tools/distrib/python/grpcio_tools/grpc_tools/command.py

```diff
@@ -63,3 +118,26 @@ def run(self):
         # to `self.distribution.package_dir` (and get a key error if it's not
         # there).
         build_package_protos(self.distribution.package_dir[''])
+
+
+class BuildPackageProtosStrict(setuptools.Command):
+    """Command to strictly generate project *_pb2.py modules from proto files.
+
+    The generation will abort if any of the proto files contains error.
+    """
+
+    description = 'strictly build grpc protobuf modules'
+    user_options = []
+
+    def initialize_options(self):
+        pass
+
+    def finalize_options(self):
+        pass
+
+    def run(self):
+        # due to limitations of the proto generator, we require that only *one*
+        # directory is provided as an 'include' directory. We assume it's the '' key
+        # to `self.distribution.package_dir` (and get a key error if it's not
+        # there).
+        build_package_protos_strict(self.distribution.package_dir[''])
```

Comment:
I'm not sure this is the best API for this. How would I know to use the empty string as a key? This will also silently fail if I offer any other package_dirs. Perhaps we could throw an error if `len(self.distribution.package_dir) != 1` and then simply use whichever key was provided. This interface seems more intuitive to me.

Reply:
That’s the design for grpc_tools and the magical key is set to “” for a long time. I don’t really like it, but I can’t come up with a backward compatible improvement.

If user is compiling with multiple package_dir, they will have no way to fix it...